icepyx/core/spatial.py

```diff
@@ -67,17 +88,29 @@ def geodataframe(extent_type, spatial_extent, file=False, xdateline=None):
                 for pair in zip(cartesian_lons, spatial_extent[1::2])
                 for item in pair
             ]
-            bbox = box(*cartesian_spatial_extent)
+            bbox = box(
+                cartesian_spatial_extent[0],
+                cartesian_spatial_extent[1],
+                cartesian_spatial_extent[2],
+                cartesian_spatial_extent[3],
+            )
```

Comment:
The typechecker needs to know there are only 4 things being passed, since the 5th argument to `box` must be a `bool`. The typechecker doesn't understand `*cartesian_spatial_extent[0:4]` to mean the same thing unfortunately.